### PR TITLE
migrate user_list_row_with_error to ts

### PR DESCRIPTION
**Pull request by dudupopkhadze**
#### Summary
- All test passing
-  No errors after make check-types

UserListRowWithError is only used in SystemUsersList componnet, where it is passed as rowComponentType prop to SearchableUserList, then  SearchableUserList passes it again to UserList as rowComponentType. So i backtracked all the props and declared  them using Typescript.

#### Ticket Link

  Fixes: https://github.com/mattermost/mattermost-server/issues/15469
  JIRA: https://mattermost.atlassian.net/browse/MM-20471

### CODE REVIEW

File: components/user_list_row_with_error/user_list_row_with_error.tsx [hunk 1]
```diff
@@ -1,50 +1,66 @@
...
+            channelMember: ChannelMembership;
+        };
     };
+    index?: number;
+    userCount?: number;
+    totalUsers?: number;
+    isDisabled?: boolean;
+}
+type State = {
+    error?: ServerError
```
Comment (devinbinnie) [inline components/user_list_row_with_error/user_list_row_with_error.tsx:48]: Nit: semicolon at the end of line

File: components/user_list_row_with_error/user_list_row_with_error.tsx [hunk 2]
```diff
@@ -63,7 +79,7 @@ export default class UserListRowWithError extends React.PureComponent {
         }
 
         // QUICK HACK, NEEDS A PROP FOR TOGGLING STATUS
-        let email = this.props.user.email;
+        let email:string|null|JSX.Element = this.props.user.email;
```
Comment (devinbinnie) [inline components/user_list_row_with_error/user_list_row_with_error.tsx:82]: You can use `React.ReactNode` to represent this instead of a union type.

**Review by devinbinnie — APPROVED** [review]
A few non-blocking comments, but otherwise LGTM!
Thanks @dudupopkhadze!

**Comment by dudupopkhadze** [comment]
@devinbinnie thanks for the feedback, fixed those

**Review by nevyangelova — APPROVED** [review]

**Review by stevemudie — APPROVED** [review]
LGTM

**Pull request merged**